client/ayon_core/hosts/resolve/plugins/load/load_media.py

```diff
@@ -0,0 +1,365 @@
+
+        # load clip to timeline and get main variables
+        project_name = get_current_project_name()
+        version = get_version_by_id(project_name, representation["parent"])
+        version_data = version.get("data", {})
+        version_name = version.get("name", None)
+        colorspace = version_data.get("colorspace", None)
+
+        # add additional metadata from the version to imprint Avalon knob
+        add_keys = [
+            "frameStart", "frameEnd", "source", "author",
+            "fps", "handleStart", "handleEnd"
+        ]
+        data = {
+            key: version_data.get(key, str(None)) for key in add_keys
+        }
+
+        # add variables related to version context
+        data.update({
+            "representation": str(representation["_id"]),
+            "version": version_name,
+            "colorspace": colorspace,
+        })
+
+        return data
+
+    @classmethod
+    def get_item_color(cls, representation) -> str:
+        """Return item color name.
+
+        Coloring depends on whether representation is the latest version.
+        """
+        # Compare version with last version
+        project_name = get_current_project_name()
+        version = get_version_by_id(
+            project_name,
+            representation["parent"],
+            fields=["name", "parent"]
+        )
+        last_version = get_last_version_by_subset_id(
+            project_name,
+            version["parent"],
+            fields=["name"]
+        ) or {}
+
+        # set clip colour
+        if version.get("name") == last_version.get("name"):
+            return cls.clip_color_last
+        else:
+            return cls.clip_color
+
+    def _set_metadata(self, media_pool_item, context: dict):
+        """Set Media Pool Item Clip Properties"""
+
+        # Set the timecode for the loaded clip if Resolve doesn't parse it
+        # correctly from the input. An image sequence will have timecode
+        # parsed from its frame range, we will want to preserve that.
+        # TODO: Setting the Start TC breaks existing clips on update
+        #   See: https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=197327
+        #   Once that's resolved we should enable this
+        # start_tc = media_pool_item.GetClipProperty("Start TC")
+        # if start_tc == "00:00:00:00":
+        #     from openpype.pipeline.editorial import frames_to_timecode
+        #     # Assume no timecode was detected from the source media
+        #
+        #     fps = float(media_pool_item.GetClipProperty("FPS"))
+        #     handle_start = context["version"]["data"].get("handleStart", 0)
+        #     frame_start = context["version"]["data"].get("frameStart", 0)
+        #     frame_start_handle = frame_start - handle_start
+        #     timecode = frames_to_timecode(frame_start_handle, fps)
+        #
+        #     if timecode != start_tc:
+        #         media_pool_item.SetClipProperty("Start TC", timecode)
```

Comment:
The workaround we discussed on discord, to in `ExtractReview` write along the `-timecode` output argument for FFMPEG works nicely for Resolve to pick up the timecode of the media, which doesn't require us to at least for those files do a "dance" like this. Which makes this much less important; and because it's currently not possible in Resolve it seems to fix this anyway via the code (as seen in [the linked thread](https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=197327) in the comment) I think we should remove this commented code.

Reply:
Here's that workaround logic I implemented for outputs from reviews: https://github.com/BigRoy/OpenPype/commit/4be1d58a5c53b6ac2640f161a6d47b77540ee44b

Then you can add it as output argument in settings (e.g. here in OpenPype):
![image](https://github.com/ynput/ayon-core/assets/2439881/e265e099-a5d6-4c4a-8b2a-6074e97de618)

Then resolve loads that natively for the clip, like seen here:
![image](https://github.com/ynput/ayon-core/assets/2439881/fd602415-f7d5-459c-9522-a1dcae18ed40)